awx/main/models/execution_environments.py

```diff
@@ -0,0 +1,41 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from awx.api.versioning import reverse
+from awx.main.models.base import PrimordialModel
+
+
+__all__ = ['ExecutionEnvironment']
+
+
+class ExecutionEnvironment(PrimordialModel):
+    class Meta:
+        unique_together = ('organization', 'image')
+        ordering = (models.F('organization_id').asc(nulls_first=True), 'image')
+
+    organization = models.ForeignKey(
+        'Organization',
+        null=True,
+        default=None,
+        blank=True,
+        on_delete=models.CASCADE,
+        related_name='%(class)ss',
+        help_text=_('The organization used to determine access to this execution environment.'),
+    )
+    image = models.CharField(
+        max_length=1024,
+        verbose_name=_('image location'),
+        help_text=_("The registry location where the container is stored."),
+    )
+    managed_by_tower = models.BooleanField(default=False, editable=False)
+    credential = models.ForeignKey(
+        'Credential',
+        related_name='%(class)ss',
+        blank=True,
+        null=True,
+        default=None,
+        on_delete=models.SET_NULL,
+    )
```

Comment:
would awx log in w/ the credential and will ansible-runner always just assume we are allready authenticated w/ the registry?

What type of registry credentials are we going to support?

Reply:
I don't know, I'm still pretty iffy on how this part will work.

Reply:
Do we have an idea of what _credential type_ this will need, and can we enforce that in the serializer or clean methods?